packages/extension-store-locator/src/components/store-locator/store-locator-form.jsx

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React, {useEffect} from 'react'
+import PropTypes from 'prop-types'
+import {useForm} from 'react-hook-form'
+import {Box} from '@chakra-ui/react/dist/cjs/box/box.cjs'
+import {Button} from '@chakra-ui/react/dist/cjs/button/button.cjs'
+import {InputGroup} from '@chakra-ui/react/dist/cjs/input/input-group.cjs'
+import {Select} from '@chakra-ui/react/dist/cjs/select/select.cjs'
+import {FormControl} from '@chakra-ui/react/dist/cjs/form-control/form-control.cjs'
+import {FormErrorMessage} from '@chakra-ui/react/dist/cjs/form-control/form-error.cjs'
+import {Input} from '@chakra-ui/react/dist/cjs/input/input.cjs'
+
+// import {AlertIcon} from '@salesforce/retail-react-app/app/components/icons'
+import {Controller} from 'react-hook-form'
+import {useStoreLocator} from '*/components/store-locator/use-store-locator'
+
+const useGeolocation = () => {
+    const {
+        setSearchStoresParams,
+        setAutomaticGeolocationHasFailed,
+        setUserHasSetManualGeolocation,
+        userHasSetManualGeolocation,
+        config
+    } = useStoreLocator()
+
+    const getGeolocationError = () => {
+        setAutomaticGeolocationHasFailed(true)
+    }
+
+    const getGeolocationSuccess = (position) => {
+        setAutomaticGeolocationHasFailed(false)
+        setSearchStoresParams({
+            latitude: position.coords.latitude,
+            longitude: position.coords.longitude,
+            limit: config.defaultPageSize
+        })
+    }
+
+    const getUserGeolocation = () => {
+        if (navigator?.geolocation) {
+            navigator.geolocation.getCurrentPosition(getGeolocationSuccess, getGeolocationError)
+            setUserHasSetManualGeolocation(false)
+        } else {
+            console.log('Geolocation not supported')
+        }
+    }
+
+    useEffect(() => {
+        if (!userHasSetManualGeolocation) getUserGeolocation()
+    }, [])
+
+    return getUserGeolocation
+}
```

Comment:
TODO (if time permits)

I don't like this code, it's tangled with other hooks and components. I'm looking to refactor it following the single responsibility principle to create a new `useGeoLocation` hook that only deals with the browser navigator api.